NotificationsPipeline.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime
+import NotificationsMessage as nm
+import importlib
+# Imports for all the pipeline steps
+from notifications_pipeline_steps.NotificationsPipelineUserTargeting import NotificationsPipelineUserTargeting
+from notifications_pipeline_steps.NotificationsPipelineUserPolicy import NotificationsPipelineUserPolicy
+from notifications_pipeline_steps.NotificationsPipelinePresentation import NotificationsPipelinePresentation
+from notifications_pipeline_steps.NotificationsPipelineDelivery import NotificationsPipelineDelivery
+
+class NotificationsPipeline(object):
+    pipeline_step_list = [
+    "NotificationsPipelineComposer",
+        "notifications_pipeline_steps.NotificationsPipelineUserTargeting",
+            "notifications_pipeline_steps.NotificationsPipelineUserPolicy",
+                "notifications_pipeline_steps.NotificationsPipelinePresentation",
+                    "notifications_pipeline_steps.NotificationsPipelineDelivery"
+    ]
+    def __init__(self):
+        pass
+    @staticmethod
+    def str2Class(module_name, class_name):
+        # Remove during refactoring
+        return getattr(importlib.import_module(module_name), class_name)
+    @staticmethod
+    def on_enter(message, pipeline_step_list = pipeline_step_list):
+        # Identify first time
+        if message.current_step == pipeline_step_list[0]:
+            message.history = [nm.NotificationsPipelineHistory() for i in range(len(pipeline_step_list) - 1)] # Initialize list of history objects
+            message.history[0].enter_timestamp = datetime.utcnow() # Assign the entry timestamp
```

Comment:
I would prefer to see history entries not get modified after they are created.  

Maybe consider a structure like:

    class History(object):
        def __init__(self, step, event):
            self.step = step
            self.event = event
            self.timestamp = datetime.utcnow()

Then instead of building up a specific-length history at the beginning, just start with an empty list, and append to it whenever new history events come in:

    message.history.append(History(message.current_step, 'started')) 
    message.perform_the_step()
    message.history.append(History(message.current_step, 'completed')) 

This will be more flexible if we want to change what kind of events we're tracking, how many steps are being performed, or anything else.